utils/environment.go

```diff
@@ -68,13 +68,13 @@ type envEnvelope struct {
 	Languages  LanguageList `json:"languages"`
 }
 
-func ReadEnvironment(data *json.RawMessage) (*environment, error) {
+func ReadEnvironment(data json.RawMessage) (*environment, error) {
```

Comment:
Sure sure - `data` here contains a pointer to the data so you don't have to worry about it passing the entire buffer as a value... so it sorta acts like passing reference. I noticed the json.Marshal/Unmarshal methods don't take pointers so I think this is more goish (?).